### PR TITLE
Provide transport info

### PR DESCRIPTION
**Pull request by de-vri-es**
This PR makes information about the connection available to users of the crate.

It is not possible to retrieve this information through a `PeerHandle`, primarily because the `PeerHandle` has no idea which underlying transport is used. However, you can get the information from a `Transport` and from a `Peer` (through `peer.transport()`) before the `Peer` is spawned.

Additionally, functions that directly create a `PeerHandle` without exposing the `Transport` also provide you with the info object. This means `Peer::connect()`, `Listener::accept()` and `Listener::run()`.

For TCP connections, the information includes a local and remote socket address.

For connection oriented Unix sockets (stream and seqpacket), the information includes a UID, GID and PID. I did not add a local or remote address for these, since connected Unix sockets have no address.

### CODE REVIEW

**Review by hgaiser — APPROVED** [review]
Why change the `Server` name to `Listener`?

**Comment by de-vri-es** [comment]
> Why change the `Server` name to `Listener`?

This actually happened much earlier, I just forgot to regenerate README.md before.

But the reason for the rename was: the `interface!` macro generates a `Client` and `Server` struct. But the `Server` only serves *one* peer. It doesn't accept multiple connections.

A socket that accepts connections is often called either a listener (or listening socket) or an acceptor. I choose to rename the old `Server` struct to `Listener` because it *only* listens for new connections and then requires the user to handle the peer (or: serve it).

So basically, to avoid confusion between the new generated `Server`, I renamed the old one with a completely different purpose to `Listener`.

I'm reasonably happy with these names, but 0.5 is still in alpha, so we can still rename both if we can come up with better names.

**Review by dan-jfisher — APPROVED** [review]
